main.lua

```diff
@@ -78,20 +70,21 @@ local availableHouses = {
     house13 = true
 }
 
+-- hack, see method comment
 increment(mapData.layers)
 
 local buyHouseButton
-local goldText
+local score
 local map
 
 -- updates players gold count to the new provided amount, and updates UI
-local function updateGold(newGold)
-    playerGold = newGold
-    goldText.text = "Gold: " .. playerGold
-    if playerGold < math.pow(2, tablelength(playerHouses)) then
-        buyHouseButton:setEnabled(false)
+local function updateGold(increment)
```

Comment:
I don't like how we constantly update buyHouseButton, is there a better way? Is there was a way to throw this inside of the button/score object instead of having it hang around in no-mans land?

Reply:
Agreed it's ugly, will fix later :)